javascript/node/selenium-webdriver/lib/virtual_authenticator.js

```diff
@@ -0,0 +1,230 @@
+// Licensed to the Software Freedom Conservancy (SFC) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The SFC licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+'use strict'
+
+/**
+ * Options for the creation of virtual authenticators.
+ * @see http://w3c.github.io/webauthn/#sctn-automation
+ */
+class VirtualAuthenticatorOptions {
+
+  static Protocol = {
+    "CTAP2": 'ctap2',
+    "U2F": 'ctap1/u2f',
+  }
+  
+  static Transport = {
+    "BLE": 'ble',
+    "USB": 'usb',
+    "NFC": 'nfc',
+    "INTERNAL": 'internal',
+  }
+
+  /**
+   * Constructor to initialise VirtualAuthenticatorOptions object.
+   */
+  constructor() {
+    this._protocol = VirtualAuthenticatorOptions.Protocol["CTAP2"]
```

Comment:
Why are we hardcoding the protocol in here ?

According to W3C the default value is none right ? Cant we pass params in constructor and make use of it for protocol and transport ?

like ,
```
this._protocol = VirtualAuthenticatorOptions.Protocol[`${protocol}`] 
```

Reply:
I think you're right. Passing the parameters in the constructor itself would be more logical. I'll fix this.

Reply:
@harsha509 Virtual Authenticator is only available for chromium bases browsers like Chrome/Egde and if we look for Virtual Authenticator under WebAuthn in `inspect > more tools > WebAuthn` we can see the default options are the one that we have in the constructor. I think that is the reason why it was hardcoded in Java. But we also have getter/setter for them if we do need to update. And i think it is a good approach. 

<img width="395" alt="Screenshot 2022-05-31 at 8 57 30 AM" src="https://user-images.githubusercontent.com/29910324/171087512-c55f3e29-34a7-437b-9390-72876d5d8d6e.png">


One can directly initialise VirtualAuthenticatorOptions with default settings. 
```python
    options = VirtualAuthenticatorOptions()
    options.protocol = VirtualAuthenticatorOptions.Protocol.U2F
    options.has_resident_key = True
    driver.add_virtual_authenticator(options)
```

Reply:
Makes sense!

I was looking it in here https://w3c.github.io/webauthn/#sctn-automation-add-virtual-authenticator. As it says default is none!

Reply:
Default is None but it is not a valid value so the user must select some value from `"ctap1/u2f", "ctap2", "ctap2_1"`. Adding a valid value as default makes sense.

Reply:
To be in sync with other language bindings implementation, keeping changes as it is!